domain_event_broker/events.py

```diff
@@ -52,24 +53,24 @@ def __init__(self,
         self.retries = retries
 
     @classmethod
-    def from_json(cls, json_data):
+    def from_json(cls, json_data: Union[bytes, str]) -> 'DomainEvent':
         """
         Create a DomainEvent from ``json_data``. Note that you probably want to
         dispatch first based on domain and event type.
 
         :param str json_data: Serialized domain event data.
         :rtype: DomainEvent
         """
-        try:
+        if isinstance(json_data, bytes):  # needed for Python 3.5
             json_data = json_data.decode('utf-8')
-        except AttributeError:
-            pass
         return cls(**json.loads(json_data))
 
-    def __repr__(self):
-        return u"DomainEvent('{0.routing_key}', '{0.data}', (domain_obj_id: {0.domain_object_id}, uuid: {0.uuid_string}))".format(self) # noqa
+    def __repr__(self) -> str:
+        return "DomainEvent('{0.routing_key}', '{0.data}', (domain_obj_id: {0.domain_object_id}, uuid: {0.uuid_string}))".format(self) # noqa
 
     __str__ = __repr__
 
-    def __eq__(self, other):
+    def __eq__(self, other: object) -> bool:
```

Comment:
Shouldn't we type `other` as `DomainEvent` here?

Reply:
I had that originally and this is what mypy had to say about it:

```
domain_event_broker/events.py:73: error: Argument 1 of "__eq__" is incompatible with supertype "object"; supertype defines the argument type as "object"
domain_event_broker/events.py:73: note: It is recommended for "__eq__" to work with arbitrary objects, for example:
domain_event_broker/events.py:73: note:     def __eq__(self, other: object) -> bool:
domain_event_broker/events.py:73: note:         if not isinstance(other, DomainEvent):
domain_event_broker/events.py:73: note:             return NotImplemented
domain_event_broker/events.py:73: note:         return <logic to compare two DomainEvent instances>
Found 1 error in 1 file (checked 24 source files)
```

I didn't see any good reasons to do it differently.

Reply:
Well, who'd want to argue with mypy...